hack/lib/log.sh

```diff
@@ -0,0 +1,216 @@
+#!/bin/bash
```

Comment:
in some future pull, this seems to be a reasonable spot for our existing log functions and green,red, etc.